src/DGtal/geometry/surfaces/DigitalSurfaceConvolver.ih

```diff
@@ -1096,15 +1096,18 @@ DGtal::DigitalSurfaceConvolver< Functor, KernelFunctor, KSpace, DigitalKernel, 2
               {
                 for( KernelConstIterator itm = myKernelMask->first; itm != myKernelMask->second; ++itm )
                   {
-                    myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
+                    if( myKSpace.cIsInside( myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel ))
+                    {
+                      myKSpace.sSetKCoords( shiftedSpel, myKSpace.sKCoords( myKSpace.sSpel( *itm )) + shiftInnerSpel );
```

Comment:
If I understood correctly, it will only affect *sometimes* the choice to reuse last result instead compute with the whole kernel. It will not affect the result provides, only the computation time.